Tensile/Source/lib/source/ContractionProblem.cpp

```diff
@@ -638,6 +635,19 @@ namespace Tensile
         if(sizeMapping.persistentKernelAlongBatch)
             problemTiles *= numWG.z;
 
+        size_t cuCount      = dynamic_cast<AMDGPU const&>(hardware).computeUnitCount;
```

Comment:
If you are not doing any nullptr checking and are fairly certain this will be a AMDGPU object, maybe consider static_cast? Is there any reason hardware obj would be something else?

If dynamic casting is still required, might be a good idea to add nullptr checks.

Reply:
Thx! agree on this.